rxlifecycle/src/main/java/com/trello/rxlifecycle/RxLifecycle.java

```diff
@@ -286,6 +327,27 @@ public FragmentEvent call(FragmentEvent lastEvent) {
             }
         };
 
+    private static final Func1<ServiceEvent, ServiceEvent> SERVICE_LIFECYCLE =
+            new Func1<ServiceEvent, ServiceEvent>() {
+                @Override
+                public ServiceEvent call(ServiceEvent lastEvent) {
+                    switch (lastEvent) {
+                        case CREATE:
+                        case START_COMMAND:
+                            return ServiceEvent.DESTROY;
+                        case BIND:
+                        case REBIND:
+                            return ServiceEvent.UNBIND;
```

Comment:
While this seems technically correct, I'm worried about the effectiveness of using RxLifecycle for bind/unbind events. Think about this process:

1. Bind 1 is called and uses this
2. Bind 2 is called and uses this
3. Bind 1 ends; subscriptions are NOT ended, since unBind hasn't been called yet
4. Bind 2 ends; NOW all subscriptions are ended.

I'm not sure this is a solvable problem with the current architecture.